### PR TITLE
added team_stats function and tests

### LINKED ISSUE #25: Team_stats function and tests for Milestone 2
This milestone requires that the team_stats function be competed and that there is at least 1 unit test covering at least 2 - 3 edge cases. The requirements are detailed as follows:

 Iteratively code up the function.
 At least 1 unit test.
 The unit test should cover at least 2 - 3 edge cases.
 The test(s) should cover all branches of program flow (i.e. both sides of an if-statement).
 Update the documentation in response to the coding and the tests for any changes.
 Document the test function(s).
 Check the that the code passes the tests.
 Write the test in the pytest module.
I have thought about the possible tests, and some ideas are listed below:

Check that no more and now fewer than 100 results are returned.
Make sure the end_date is at least equal to or grater than the start_date.
Ensure the start_date and end_date are of the appropriate type.
Make sure the start_date and end_date are valid dates to query.
Update the documentation describing what the valid dates are.
Make sure the default values for the current season are appropriate (i.e. if the arguments are left blank).
Check that the function returns a dataframe object.
Check that the API request returns a valid response.
This could either be an assert, and/or a try/catch statement.
Implement a system to handle when bad responses are returned.
Identify why a bad response was made.
Test that the URL is in-fact correctly specified.

### PR DESCRIPTION
**Pull request by ManishPJoshi**
can you please approve

### CODE REVIEW

File: tests/test_pypuck.py [hunk 1]
```diff
@@ -61,5 +61,23 @@ def test_player_stats_bad(start_date='2020-02-28', end_date='2019-10-02'):
...
+		end_date {str} -- end_date to query (default: {'20192020'})
+
+	Raises:
+		TypeError: A message if return type is wrong (wrong data type).
+		ValueError: A message if return value is wrong (not enough data).
+	"""
+	# Test for various end_dates
+	df = pypuck.team_stats(start_season='19992000',end_season='20102011')
+	if df.empty== True:
+		raise TypeError("invalid Inputs, Season_start should be later than Season end. Valid seasons are from 1917 to 2020")
```
Comment (jnederlo) [inline tests/test_pypuck.py]: You are missing a space around `==`. Because this is a new branch I'm not able to pull it to test this. And it would appear that this hasn't been tested.

```suggestion
	if df.empty == True:
		raise TypeError("invalid Inputs, Season_start should be later than Season end. Valid seasons are from 1917 to 2020")
```
@ManishPJoshi please also add in another test to test. For example, you could test that the function returns a `type` `pd.DataFrame`. You could also check that the number of rows in the returned df is proper, and that the values do in-fact range from the 19992000 season to the 20102011 season, or easier, if you sent in no arguments, that the function returned data from the current year.

File: pypuck/pypuck.py [hunk 2]
```diff
@@ -139,7 +139,26 @@ def team_stats(start_season=None, end_season=None):
...
+      print("Invalid Input. Enter end_season in Bi-Annual format. 'YYYYYYYY'")
+      return
+    # print(base_url + arguments)
+    page = requests.get(base_url + arguments)
+    #print(page.url)
+    df = pd.DataFrame(page.json()['data'])
+    if df.empty == True:
+      print("Invalid Inputs. Season_start should be later than Season end. Valid seasons are from 1917 to 2020. Enter years in Bi-Annual format of 'YYYYYYYY'")
+    else:
+      print(df.head())
```
Comment (jnederlo) [inline pypuck/pypuck.py]: We probably don't want to include this test here. Instead we will cover it in the `test_pypuck.py` module, which you have already done, so it can be removed. Additionally, we shouldn't print the head.

```suggestion
```

File: pypuck/pypuck.py [hunk 3]
```diff
@@ -139,7 +139,26 @@ def team_stats(start_season=None, end_season=None):
...
+          f' and seasonId>={start_season}'
+    if str.isdecimal(start_season) == False:
+      print("Invalid Input. Enter start_season in Bi-Annual format. 'YYYYYYYY'")
+      return
+    if str.isdecimal(end_season) == False:
+      print("Invalid Input. Enter end_season in Bi-Annual format. 'YYYYYYYY'")
+      return
+    # print(base_url + arguments)
+    page = requests.get(base_url + arguments)
+    #print(page.url)
```
Comment (jnederlo) [inline pypuck/pypuck.py]: No need to have these commented out lines remain in the code. Also, please add in test from helpers checking if status code is proper.

```suggestion

    page = requests.get(base_url + arguments)

    # Check the response code is valid - i.e. the API didn't fail
    helpers.check_response_code(page.status_code)
```

File: pypuck/pypuck.py [hunk 4]
```diff
@@ -139,7 +139,26 @@ def team_stats(start_season=None, end_season=None):
...
+    base_url = 'https://api.nhle.com/stats/rest/en/team/summary?'
+    arguments = 'cayenneExp=gameTypeId=2' +\
+          f' and seasonId<={end_season}' +\
+          f' and seasonId>={start_season}'
+    if str.isdecimal(start_season) == False:
+      print("Invalid Input. Enter start_season in Bi-Annual format. 'YYYYYYYY'")
+      return
+    if str.isdecimal(end_season) == False:
+      print("Invalid Input. Enter end_season in Bi-Annual format. 'YYYYYYYY'")
+      return
```
Comment (jnederlo) [inline pypuck/pypuck.py]: I made a helper function that can handle this error, so you can add this to the code instead.

```suggestion
    helpers.check_argument_type(start_season, 'start_season', str)
    helpers.check_argument_type(end_season, 'end_season', str)
```

**Review by jnederlo — CHANGES_REQUESTED** [review]

File: tests/test_pypuck.py [hunk 5]
```diff
@@ -61,5 +61,23 @@ def test_player_stats_bad(start_date='2020-02-28', end_date='2019-10-02'):
...
+	"""
+	Test function to check proper inputs and returns.
+
+	Keyword Arguments:
+		start_date {str} -- start_date to query (default: {'20192020'})
+		end_date {str} -- end_date to query (default: {'20192020'})
+
+	Raises:
+		TypeError: A message if return type is wrong (wrong data type).
+		ValueError: A message if return value is wrong (not enough data).
```
Comment (jnederlo) [inline tests/test_pypuck.py]: @ManishPJoshi This docstring was copied from my function. Please add in an appropriate docstring for your function. For example:

```suggestion
	Function to test that the team_stats function appropriately returns a pd.DataFrame object and that the resulting dataframe has the queried data. 

	Keyword Arguments:
		start_season {str} -- start_season to query (default: {'20192020'})
		end_season {str} -- end_season to query (default: {'20192020'})

	Raises:
		ValueError: A message if return value is wrong (empty return).
```

File: pypuck/pypuck.py [hunk 6]
```diff
@@ -157,7 +157,7 @@ def team_stats(start_season= "20192020", end_season="20192020"):
     if df.empty == True:
       print("Invalid Inputs. Season_start should be later than Season end. Valid seasons are from 1917 to 2020. Enter years in Bi-Annual format of 'YYYYYYYY'")
     else:
```
Comment (jnederlo) [inline pypuck/pypuck.py]: As I previously pointed out, this is not proper and shouldn't be included.

```suggestion
```

File: pypuck/pypuck.py [hunk 7]
```diff
@@ -157,7 +157,7 @@ def team_stats(start_season= "20192020", end_season="20192020"):
...
-      return df.columns
+      return df
```
Comment (jnederlo) [inline pypuck/pypuck.py]:
```suggestion
return df
```

**Review by jnederlo — APPROVED** [review]
